### PR TITLE
Prevent setting a strategy on `swarm update <service>`

### PR DESCRIPTION
**Pull request by ewoutp**
This change gives an error if the use specifies an update strategy while trying to update a service.

```
$ swarm update -s one-by-one swarm.json
Strategies are not supported for updating a service.
```

See giantswarm/app-service#472

### CODE REVIEW

**Comment by ewoutp** [comment]
@teemow @marians RFR

File: update.go [hunk 1]
```diff
@@ -115,7 +115,10 @@ func updateCmdRun(cmd *cobra.Command, args []string) {
 	}
 }
 
-func updateService(definitionFilePath string) error {
+func updateService(cmd *cobra.Command, definitionFilePath string) error {
+	if cmd.Flag("strategy").Changed {
+		Quitf("Strategies are not supported for updating a service.\n")
```
Comment (teemow) [inline update.go]: Afaik 'hot-swap' is the strategy that is used.

**Comment by teemow** [comment]
Another thing would be to work on the help output. It is currently not really clear for the user, that there is a difference between `swarm update` and `swarm update component version`. 

```
➜ swarm update --help
Update a service by changing its definition or update a component by changing the docker image to a newer version

Usage: 
  swarm update [flags]

Docs:
  https://docs.giantswarm.io/reference/cli/update/

Examples:
  -y <swarm.json>                                                       Update a service by changing its definition. Use -y to accept
  -d <swarm.json>                                                       Update a service by changing its definition. Use -d to detach
  -d <service>/<component>                                              Update a component to the 'latest' version. Use -d to detach
  -d <service>/<component> <version>                                    Update a component using the given docker image version. Use -d to detach
  -d -s=one-by-one|all-at-once|hot-swap <service>/<component> <version> Update a component using the given docker image version. Use -d to detach. Use -s to choose an update strategy

Flags:
  -d, --detach[=false]: Do not wait for the component to be updated
  -s, --strategy="one-by-one": 
      --var=[]: Key-Value variable pair, used in swarm.json
      --var-file="": Path to a file containing key-value variable pairs, used in swarm.json
  -y, --yes[=false]: Accept the plan to update the service

Global Flags:
      --api-endpoint="https://api.giantswarm.io/v1": The API endpoint to talk to
      --config-dir="": Change the default configuration directory
      --debug[=false]: Log debug output or not
      --env="giantswarm/production": Which environment to use. Format: 'organization/environment'
  -q, --quiet[=false]: Suppress output
  -v, --verbose[=false]: Verbose output
```

File: update.go (hunk 1 above)
Comment (ewoutp) [inline update.go]: Correct, but currently that is an "implementation detail". Do you want to visible here?
I'm open to suggestion for better feedback text. @marians

Comment (teemow) [inline update.go]: As a user I find this really important. Otherwise I have no clue what is going to happen.

**Comment by ewoutp** [comment]
ping @marians

**Comment by marians** [comment]
As far as I understand, we currently use one default strategy for updating a service. Don't know which one it is, assuming hot-swap.

So if the user does a `swarm update -s hot-swap swarm.json`, we should actually handle this call just fine and print an additional info (to STDERR?):

    Hint: Selecting a strategy when updating a service definition is not possible,
    since the 'hot-swap'  strategy will be used in any case.

**Comment by ZeissS** [comment]
@zyndiecate can you review this? :P

**Comment by ewoutp** [comment]
@marians I told you that I would implement your suggested hint, but I'm having second thoughts.

If we build in knowledge in the CLI about the currently used update strategy we would have release a new cli if we change the service update implementation. Because of the volume discussion I think it is likely that we will not use hot-swap updates in all cases soon.

Therefore I think it may be better to avoid accepting a strategy all together.
What do you think?

**Comment by marians** [comment]
@ewoutp Makes sense. Then skip this. The rest looks good to me.

**Pull request merged**